rasa/importers/importer.py

```diff
@@ -271,3 +274,118 @@ async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
         return reduce(
             lambda merged, other: merged.merge(other), nlu_data, TrainingData()
         )
+
+
+class E2EImporter(TrainingDataImporter):
+    """Importer which
+    - enhances the NLU training data with actions / user messages from the stories.
+    - adds potential end-to-end bot messages from stories as actions to the domain
+    """
+
+    def __init__(self, importer: TrainingDataImporter) -> None:
+        self._importer = importer
```

Comment:
why importer has `_importer` attribute?

Reply:
Because we are using the [composite pattern](https://en.wikipedia.org/wiki/Composite_pattern) here. This code was actually already reviewed by Rick (https://github.com/RasaHQ/rasa/pull/6083)